m2cgen/ast.py

```diff
@@ -54,10 +56,8 @@ class NumExpr(Expr):
 
 
 class NumVal(NumExpr):
-    def __init__(self, value, dtype=None):
-        if dtype:
-            value = dtype(value)
-        self.value = value
+    def __init__(self, value, dtype=np.float64):
+        self.value = dtype(value)
```

Comment:
Some languages prohibit implicit type casting like F#. Therefore, we should ensure that each value is double unless we explicitly specify another type. Refer to https://stackoverflow.com/questions/45710437/why-cant-you-add-int-and-float-literals-in-f.